text/0000-grid-primitives.md

```diff
@@ -0,0 +1,760 @@
+property definitions which minimally include the property’s type (integer,
+string, enum, etc.). In addition, the properties (for example, product
+description, GPS lat/long, or product dimensions) should always be stored and
+exchanged using the same format within Grid components.
+
+# Guide-Level Explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## PropertyDefinition
+
+A property is defined using a `PropertyDefinition` which includes the following:
+
+- Data type (one of: BYTES, BOOLEAN, NUMBER, STRING, ENUM, STRUCT, LAT_LONG,
+  DATETIME)
+- Name
+- Type description
+- Optionality (whether or not the field is required)
+
+## Data types
+
+### Bytes
+
+A Bytes data type is an array of raw bytes.  This can be used to store
+arbitrary, opaque data. For example, a property with the Bytes data type could
+be used to store serialized JSON objects containing application metadata for a
+field, such as an image URL or style name.
+
+### Booleans
+
+A boolean data type restricts a value to True and False. Though boolean types
+could be stored in other integer (or byte) types using 0 or 1, an explicit
+boolean type assists in capturing intent and restricting the value.
+
+### Strings
+
+A string data type contains a standard UTF-8 encoded string value.
+
+### Numbers
+
+Numbers are represented as an integer with a given precision.  This can be
+thought of as akin to scientific notation. An instance of a number with this
+property definition is represented as a value (the significand) with the
+exponent (the order of magnitude) defined in the schema itself. So for example:
+
+```
+(value: 24, exponent: 3)  -> 24 * 10^3  -> 24000
+(value: 24, exponent: -3) -> 24 * 10^-3 -> 0.024
+(value: 24, exponent: 0)  -> 24 * 10^0  -> 24
+```
+
+Importantly, this exponent will be set on a Property's schema, not when the
+value is actually input. It will affect the semantic meaning of integers stored
+under a Property, not any of the actual operations done with them. Properties
+with an exponent of 3 or -3 will always be expressed as a whole integer of
+thousands or thousandths. For this reason, the exponent should be thought of
+more as a unit of measure than as true scientific notation. 
+
+Standard integers are represented with the exponent set to zero. 
+
+### Enums
+
+An enum data type restricts values to a limited set of possible values. The
+definition for this data type includes a list of strings names describing a
+possible state of the enum. 
+
+### Structs
+
+A struct is a recursively defined collection of other named properties that
+represents two or more intrinsically linked values, like X/Y coordinates or RGB
+colors. These values can be of any Supply Chain data type, including STRUCT,
+allowing nesting to an arbitrary depth. Although versatile and powerful, structs
+are heavyweight and should be used conservatively; restrict struct use to
+linking values that must always be updated together. The transaction processor
```

Comment:
Should probably say "smart contract" here instead of "transaction processor".

Reply:
Removed two occurrances in a new commit.